apps/prairielearn/python/test/core_test.py

```diff
@@ -160,7 +162,7 @@ def test_networkx_serialization(networkx_graph: Any) -> None:
     json_object = json.dumps(pl.to_json(networkx_graph), allow_nan=False)
     decoded_json_object = pl.from_json(json.loads(json_object))
 
-    assert type(networkx_graph) == type(decoded_json_object)  # noqa: E721
+    assert isinstance(decoded_json_object, type(networkx_graph))
```

Comment:
I'm not sure we want this change? Iirc the networkx graphs inherit from each other, and we want to assert the type is preserved exactly.

Reply:
You are right, pyright was having issues. I added a note about this in [935d5ad](https://github.com/PrairieLearn/PrairieLearn/pull/11192/commits/935d5adbb614ccdd78e593685c4762838bdaa0c8).